pliers/extractors/audio.py

```diff
@@ -488,4 +494,88 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
+        returned. If not defined, returns probabilities for all labels.
+    spectrogram (bool): if True, plots mel-spectrogram of the audio stimulus.
+    yamnet_kwargs (optional): Optional named arguments that modify input 
+        parameters for the model (see params.py file in yamnet repository)
+    '''
+
+    _log_attributes = ('params', 'top_n')
+
+    def __init__(self, hop_size=0.1, top_n=None, 
+                 spectrogram=False, **yamnet_kwargs):
+        verify_dependencies(['yamnet'])
+        verify_dependencies(['tensorflow'])
+
+        MODULE_PATH = path.dirname(yamnet.__file__)
+        WEIGHTS_PATH = path.join(MODULE_PATH, 'yamnet.h5')
+        LABELS_PATH = path.join(MODULE_PATH, 'yamnet_class_map.csv')
+
+        self.spectrogram = spectrogram
+        self.params = yamnet.params
+        self.params.PATCH_HOP_SECONDS = hop_size
+
+        if yamnet_kwargs:
+            for par in self.params.__dict__:
+                if yamnet_kwargs[par]:
+                    setattr(self.params, par, yamnet_kwargs[par])
+
+        self.tf_graph = tf.Graph()
+        with self.tf_graph.as_default():
+            self.model = yamnet.yamnet_frames_model(self.params)
+            self.model.load_weights(WEIGHTS_PATH)
+
+        self.labels = pd.read_csv(LABELS_PATH)['display_name'].tolist()
+        self.params = {par: val for par,val in self.params.__dict__.items()
+                       if par.isupper()}
+        self.top_n = top_n if top_n else len(self.labels)
+        super(AudiosetLabelExtractor, self).__init__()
+
+    def _extract(self, stim):
+        
+        data = stim.data
+        self.params['SAMPLE_RATE'] = stim.sampling_rate
+        
+        with self.tf_graph.as_default():
+            preds, spectrogram = self.model.predict(np.reshape(data, [1,-1]), 
+                                                    steps=1)
+        
+        if self.spectrogram:
+            import matplotlib.pyplot as plt
+            plt.imshow(spectrogram.T, aspect='auto', interpolation='nearest', 
+                       origin='lower', cmap='RdYlBu_r')
+            plt.xlabel('Time')
+            plt.ylabel('Frequency')
+            plt.colorbar()
+            plt.show()
+        
+        idx = np.mean(preds,axis=0).argsort()
+        preds = np.flip(preds[:,idx],axis=0)[:,:self.top_n]
+        self.labels = [self.labels[i] for i in idx][::-1][:self.top_n]
+
+        durations = self.params['PATCH_HOP_SECONDS']
+        onsets = np.arange(start=0, stop=stim.duration, step=durations)
+        onsets = onsets[onsets + self.params['PATCH_WINDOW_SECONDS'] < stim.duration]
+
+        return ExtractorResult(preds, stim, self, features=self.labels,
+                               onsets=onsets, durations=durations,
+                               orders=list(range(len(onsets))))
+
+# Add pointer to installation instructions (install models, run test install - link, add to pythonpath)
+# Add length warning
```

Comment:
We discussed this, but if there was a mixin to batch the input, that would be great. Or if it was just done within here. As a user, it seems annoying to have to manually filter, especially since the results can be stiched back together after batching.